tests/protocol/test_wire.py

```diff
@@ -705,6 +756,38 @@ def http_get_handler(url, *_, **kwargs):
 
             self.assertEquals(HostPluginProtocol.is_default_channel(), False)
 
+    def test_fetch_manifest_should_raise_if_both_channels_return_an_unsuccessful_response(self):
```

Comment:
New test. This test would fail with the old code (tested it) since the fetch_manifest function would return the failed response thinking it's a valid manifest.